tokio/src/sync/broadcast.rs

```diff
@@ -1402,22 +1464,38 @@ where
 
 impl<'a, T> Drop for Recv<'a, T> {
     fn drop(&mut self) {
```

Comment:
It occurred to me that, technically, both this and previous implementation of `Recv` destructor violates Rust aliasing rules: we have a mutable reference to `self` that contains `waiter`, and even though it is an UnsafeCell, concurrently modifying it's contents violates aliasing (as explicitly stated in UnsafeCell docs). Only after we synchronize via Mutex/RwLock/Atomic can we guarantee that our mutable reference is indeed exclusive. Is this right?

Reply:
I think I found the answer [here](https://users.rust-lang.org/t/raw-pointer-aliasing-while-destructor-is-running/92587/7).

Reply:
See https://gist.github.com/Darksonn/1567538f56af1a8038ecc3c664a42462 and https://github.com/rust-lang/rfcs/pull/3467